pkg/networking/pod_eni_info_resolver.go

```diff
@@ -388,33 +401,39 @@ func (r *defaultPodENIInfoResolver) isPodSupportedByNodeENI(pod k8s.PodInfo, nod
 	return false
 }
 
-// classifyPodsByComputeType classifies in to ec2 and fargate groups
-func (r *defaultPodENIInfoResolver) classifyPodsByComputeType(ctx context.Context, pods []k8s.PodInfo) ([]k8s.PodInfo, []k8s.PodInfo, error) {
+// classifyPodsByComputeType classifies in to ec2, fargate and sagemaker-hyperpod groups
+func (r *defaultPodENIInfoResolver) classifyPodsByComputeType(ctx context.Context, pods []k8s.PodInfo) ([]k8s.PodInfo, []k8s.PodInfo, []k8s.PodInfo, error) {
```

Comment:
Can you refactor this to be more type friendly? E.g. it's non-trivial to remember arg1 is ec2, arg2 is fargate, arg3 is hyperpod.

Can you introduce a new type that basically is 

struct{
  ec2Pods []k8s.PodInfo
  fargatePods []k8s.PodInfo
  hyperpodPod []k8s.PodInfo
}

this will make the code cleaner and as we add more compute types it would be more extensible.

Reply:
yup. refactored in the last commit. thanks!